### PR TITLE
remove sexplib and ppx_sexp_conv and ppx_cstruct dependencies

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

File: lib/tX.ml [hunk 1]
```diff
@@ -72,25 +72,36 @@ module Request = struct
 end
 
 module Response = struct
-  [%%cenum
   type status =
     | DROPPED [@id 0xfffe]
     | ERROR   [@id 0xffff]
     | OKAY    [@id 0]
     | NULL    [@id 1]
```
Comment (reynir) [inline lib/tX.ml:79]: Maybe remove `[@id 0xfffe]` annotations? Or make them into comments at least.

**Review by reynir — APPROVED** [review]

File: lib/tX.ml (hunk 1 above)
Comment (hannesm) [reply lib/tX.ml:79]: done in 10abed0

File: lib/tX.ml [hunk 2]
```diff
@@ -26,22 +24,24 @@ module Request = struct
...
+  let set_req_gref c gref = Cstruct.LE.set_uint32 c 0 gref
+  let get_req_offset c = Cstruct.LE.get_uint16 c 4
+  let set_req_offset c off = Cstruct.LE.set_uint16 c 4 off
+  let get_req_flags c = Cstruct.LE.get_uint16 c 6
+  let set_req_flags c flags = Cstruct.LE.set_uint16 c 6 flags
+  let get_req_id c = Cstruct.LE.get_uint16 c 8
+  let set_req_id c id = Cstruct.LE.set_uint16 c 6 id
+  let get_req_size c = Cstruct.LE.get_uint16 c 8
+  let set_req_size c size = Cstruct.LE.set_uint16 c 8 size
+  let sizeof_req = 10
```
Comment (palainp) [inline lib/tX.ml:44]: I'm a bit late here, but it should be `  let sizeof_req = 10`.

Comment (hannesm) [reply lib/tX.ml:44]: oops, there are a couple of offsets wrong. sorry

**Pull request merged**